src/sinks/file/mod.rs

```diff
@@ -1,19 +1,29 @@
-#![cfg(feature = "disabled")]
-
-mod file;
-
-use self::file::File;
+use crate::expiring_hash_map::ExpiringHashMap;
 use crate::{
-    event::Event,
+    event::{self, Event},
     sinks::util::SinkExt,
     template::Template,
     topology::config::{DataType, SinkConfig, SinkContext},
 };
+use async_trait::async_trait;
 use bytes::Bytes;
-use futures01::{future, Async, AsyncSink, Future, Poll, Sink, StartSend};
+use futures::channel::mpsc::Receiver;
+use futures::{
+    future::{select, Either},
+    stream::StreamExt,
+};
 use serde::{Deserialize, Serialize};
-use std::{collections::HashMap, time::Instant};
-use tokio::timer::Delay;
+use std::time::{Duration, Instant};
+use tokio02::{
+    fs::{self, File},
+    io::AsyncWriteExt,
+};
+
+mod bytes_path;
+use bytes_path::BytesPath;
+
+mod streaming_sink;
```

Comment:
We could probably place this stuff in `sinks/mod.rs`

Reply:
If we end up going with it - definitely! I think we can extract `bytes_path` into global utils too. I'll do a final pass on the locations of things before the merge.

Reply:
Moving it upper now.

Reply:
Aborting, leaving it as-is for now, we'll correct it later.